README.md

```diff
@@ -120,17 +120,23 @@ name: "example rule"
 # exist, the rule applies to every pull request.
 if:
   # "changed_files" is satisfied if any file in the pull request matches any
-  # regular expression in the list.
+  # regular expression in the list. The "ignore" list is optional and will
+  # ignore any files that match the regular expressions listed.
   changed_files:
     paths:
       - "config/.*"
       - "server/views/.*\\.tmpl"
+    ignore:
+      - "config/special\\.file"
 
   # "only_changed_files" is satisfied if all files changed by the pull request
-  # match at least one regular expression in the list.
+  # match at least one regular expression in the list. The "ignore" list is
+  # optional and will ignore any files that match the regular expressions listed.
```

Comment:
Thinking about this now, I can see this behaving in two ways and I'm not sure which is correct:

1. Files in the `ignore` list are completely ignored - it doesn't matter whether they changed or not for the purposes of the rule. This is what you implemented.

2. Files in the `ignore` list are removed from the set of files that are allowed to change, i.e. changes to them cause the predicate to fail.

I think (2) is the closer behavior semantically (not in implementation) to `changed_files`. In the example below, if I make a PR that changes `config/server.yml` and `config/special.file`, should this rule apply?

Right now, I believe the predicate is true because `config/special.file` is ignored, meaning I only changed files in `config/`. This makes it no different from only specifying `paths`.

If the intent is that you ignore things that would otherwise match patterns in `paths`, then I think we need to count "ignored" files as if they are other changed files in the PR.

Or we remove the `ignore` flag from `only_changed_files`.